### ISSUE TITLE
Enabling a websocket listener slows the broker down

### ISSUE THREAD

**Issue by Andreyooo** [eclipse/mosquitto#1804]
My default listener uses tls, when I enable an extra listener with mqtt protocol, things work fine. When I switch the protocol to websockets my broker slows down. I use mosquitto version 1.6.7.

**Default listener**

![SinusNormal](https://user-images.githubusercontent.com/38321084/91449856-e7744680-e87b-11ea-8eee-52172a77aeea.PNG)

**Default listener (with extra websocket listener enabled)**
![SinusBroken](https://user-images.githubusercontent.com/38321084/91449904-f78c2600-e87b-11ea-8078-de74730a9ddd.PNG)

Also the startup logs of mosquitto are slower printed and connecting to the broker via mqtt-explorer is slower. Any idea what produces this issue?

**Comment by Andreyooo** [comment]
So I cannot reproduce this on Windows. This seems having to do with our linux build, will look further into it. Do the linux binaries have websocket support enabled or is it just the windows installer?

**Comment by Andreyooo** [comment]
Thx ralight! I will try it next week.

**Comment by jpnurmi** [comment]
Thanks for the tip about libwebsockets!

I run into this issue after enabling Mosquitto's websockets support on a Yocto Zeus -based embedded device. The device had Mosquitto 1.6.7 and libwebsockets 3.20.0 installed. I tried first upgrading to Mosquitto 1.6.12 from meta-oe master, but that didn't help. Then I found this ticket and after upgrading to libwebsockets 4.0.20, also from meta-oe master, the laggy behavior vanished.
